From 1a18a179b5523cd32b58664bcef041adc4ebe951 Mon Sep 17 00:00:00 2001
From: Beleswar Padhi <b-padhi@ti.com>
Date: Tue, 18 Mar 2025 16:19:18 +0530
Subject: [PATCH 1/2] Add TDA54 IPC support for M55 and C76 remotecores

Add new rproc ids for M55 and C76 cores and create a map with these
rproc ids. Register the map in the SOC FAMILY info.

Signed-off-by: Beleswar Padhi <b-padhi@ti.com>
---
 include/rproc_id.h |  2 ++
 src/soc.c          | 13 ++++++++++++-
 2 files changed, 14 insertions(+), 1 deletion(-)

diff --git a/include/rproc_id.h b/include/rproc_id.h
index b79e823..f67a2ce 100644
--- a/include/rproc_id.h
+++ b/include/rproc_id.h
@@ -71,6 +71,8 @@ enum rproc_id {
 	DSP_C71_2 = 13,
 	DSP_C71_3 = 14,
 	R5F_WKUP0_0 = 15,
+	M55_MAIN_0 = 16,
+	DSP_C76_0 = 17,
 	RPROC_ID_MAX,
 };
 
diff --git a/src/soc.c b/src/soc.c
index ec8829d..48c1cdd 100644
--- a/src/soc.c
+++ b/src/soc.c
@@ -42,7 +42,7 @@
 #include "rpmsg_char_internal.h"
 
 /* Increment this whenever new SoC data is added */
-#define NUM_SOC_FAMILY 10
+#define NUM_SOC_FAMILY 11
 
 struct soc_data {
 	const char *family_name;
@@ -143,6 +143,12 @@ const struct rproc_map j722s_map[] = {
 	{ .id = DSP_C71_1,   .rproc_name = "7e200000.dsp",   },
 };
 
+/* TI TDA54 SoCs */
+const struct rproc_map tda54_map[] = {
+	{ .id = M55_MAIN_0, .rproc_name = "24000000.m55", },
+	{ .id = DSP_C76_0, .rproc_name = "64000000.c76", },
+};
+
 const struct soc_data socs[NUM_SOC_FAMILY] = {
 	{
 		.family_name = "AM65X",
@@ -194,6 +200,11 @@ const struct soc_data socs[NUM_SOC_FAMILY] = {
 		.map = j722s_map,
 		.num_rprocs = (sizeof(j722s_map) / sizeof(struct rproc_map)),
 	},
+	{
+		.family_name = "TDA54",
+		.map = tda54_map,
+		.num_rprocs = (sizeof(tda54_map) / sizeof(struct rproc_map)),
+	},
 };
 
 int _rpmsg_char_find_soc_family(const char *name, struct soc_rprocs *soc)
-- 
2.34.1

